api/envoy/admin/v2alpha/config_dump.proto

```diff
@@ -0,0 +1,132 @@
+// any CLI overrides that were merged. Bootstrap configuration information can be used to recreate
+// an Envoy configuration by reusing the output as the bootstrap configuration for another Envoy.
+message BootstrapConfigDump {
+  envoy.config.bootstrap.v2.Bootstrap bootstrap = 1 [(gogoproto.nullable) = false];
+}
+
+// Envoy's listener manager fills this message with all currently known listeners. Listener
+// configuration information can be used to recreate an Envoy configuration by populating all
+// listeners as static listeners or by returning them in a LDS response.
+message ListenersConfigDump {
+  // This is the top level, "transactional," version information in the last processed LDS
+  // discovery response. If there are only static bootstrap listeners, this field will be "".
+  string version_info = 1;
+
+  // Describes a dynamically loaded cluster via the LDS API.
+  message DynamicListener {
+    // This is the per-resource version information. This version is currently taken from the
+    // transactional version_info field at the time that the listener was loaded. In the future,
+    // discrete per-listener versions may be supported by the API.
+    string version_info = 1;
+
+    // The listener config.
+    envoy.api.v2.Listener listener = 2;
+  }
+
+  // The statically loaded listener configs.
+  repeated envoy.api.v2.Listener static_listeners = 2 [(gogoproto.nullable) = false];
+
+  // The dynamically loaded active listeners These are listeners that are available to service
+  // data plane traffic.
+  repeated DynamicListener dynamic_active_listeners = 3 [(gogoproto.nullable) = false];
+
+  // The dynamically loaded warming listeners These are listeners that are currently undergoing
+  // warming in preparation to service data plane traffic. Note that if attempting to recreate an
+  // Envoy configuration from a configuration dump, the warming listeners should generally be
+  // discarded.
+  repeated DynamicListener dynamic_warming_listeners = 4 [(gogoproto.nullable) = false];
+
+  // The dynamically loaded draining listeners These are listeners that are currently undergoing
+  // draining in preparation to stop servicing data plane traffic. Note that if attempting to
+  // recreate an Envoy configuration from a configuration dump, the draining listeners should
+  // generally be discarded.
+  repeated DynamicListener dynamic_draining_listeners = 5 [(gogoproto.nullable) = false];
+}
+
+// Envoy's cluster manager fills this message with all currently known clusters. Cluster
+// configuration information can be used to recreate an Envoy configuration by populating all
+// clusters as static clusters or by returning them in a CDS response.
+message ClustersConfigDump {
+  // This is the top level, "transactional," version information in the last processed CDS
+  // discovery response. If there are only static bootstrap clusters, this field will be "".
+  string version_info = 1;
```

Comment:
In a world of incremental xDS, this transaction value might only refer to a CDS response with a single cluster, with no bearing on the other clusters that are present. Am I reading this right?

Reply:
Yes correct. It's basically the last received response in the top level version_info field (per my suggestion of keeping that field a singleton and adding per-version resources additionally). If a deployment doesn't care about this they can leave it blank.

Reply:
We can discuss at the meeting today, but I do have some concern about this being racy.

Reply:
@htuch @nt my plan after our meeting is to not really change this proto, but to change the comments a bit. I will do that sometime this week. Let me know if that sounds wrong.

Reply:
SGTM, @nt is also planning on updating the doc based on the meeting.

Reply:
@mattklein123 do you still have other work planned here to followup on comments?

Reply:
Yes I will update comments today.